addons/chemical/functions/fnc_showChemDetector.sqf

```diff
@@ -47,30 +47,25 @@ private _exposure = _display displayCtrl 18805;
         _pfhID call CBA_fnc_removePerFrameHandler;
     };
 
-    private _hour = floor dayTime;
-    private _minute = floor ((dayTime - _hour) * 60);
-
-    _time ctrlSetText (format ["%1:%2", [_hour, 2] call CBA_fnc_formatNumber, [_minute, 2] call CBA_fnc_formatNumber]);
-
     private _intensity = _unit getVariable [QGVAR(areaIntensity), 0];
 
-    _exposure ctrlSetText (_intensity toFixed 2);
+    if ((_unit getVariable [QGVAR(detectorEnabled), false])) then {
 
-    if (_intensity > 0) then {
-        private _lastSoundTime = _unit getVariable QGVAR(lastSoundTime);
-        _unit setVariable [QGVAR(areaIntensity), 0, true];
+        private _hour = floor dayTime;
+        private _minute = floor ((dayTime - _hour) * 60);
+    
+        _time ctrlSetText (format ["%1:%2", [_hour, 2] call CBA_fnc_formatNumber, [_minute, 2] call CBA_fnc_formatNumber]);
 
-        if (isNil "_lastSoundTime") then {
-            TRACE_1("undefined lastSoundTime: setting to current time",_lastSoundTime);
-            _unit setVariable [QGVAR(lastSoundTime), CBA_missionTime];
-        };
+        _exposure ctrlSetText (_intensity toFixed 2);
+
+        _unit setVariable [QGVAR(areaIntensity), _intensity, true];
 
-        if (CBA_missionTime - _lastSoundTime > 6) exitWith {
-            playSound3D [QPATHTOF(audio\chemDetector.ogg), _unit, false, getPosASL _unit, 4, 1, 10];
-            _unit setVariable [QGVAR(lastSoundTime), CBA_missionTime];
+        if (_intensity < 0.1) then {
```

Comment:
why do we set intensity like that?

Reply:
Weren't you having issues with _intensity not being fully?